### PR TITLE
inchi: init at 1.05

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
Adds InChI library for cheminformations (https://www.inchi-trust.org/)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/inchi/default.nix
Comment (markuskowa) [inline pkgs/development/libraries/inchi/default.nix:15]: If you just put `cd ./INCHI_API/libinchi/gcc` in `preBuild` then you don't need to overwrite the `buildPhase`.

File: pkgs/development/libraries/inchi/default.nix
Comment (markuskowa) [inline pkgs/development/libraries/inchi/default.nix:42]: `version` is not needed in `meta`.

File: pkgs/development/libraries/inchi/default.nix
Comment (markuskowa) [inline pkgs/development/libraries/inchi/default.nix:46]: There should also be a maintainer field.

**Review by markuskowa — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/inchi/default.nix
Comment (rmcgibbo) [reply pkgs/development/libraries/inchi/default.nix:46]: The other PR (nice-dcv-client #103939) that you just reviewed includes adding me to the maintainers list, so I'll fix this one up according to your suggestions (and add myself as maintainer) after we finish & merge that one.

**Comment by rmcgibbo** [comment]
@markuskowa: I've made the requested fixes and force-pushed.

**Review by markuskowa — APPROVED** [review]

**Pull request merged**
